src/Plate/coordinate.ts

```diff
@@ -0,0 +1,131 @@
+import { CoordinateSystem } from './coordinateSystem';
+import { Coordinates, FlowDirection } from './types';
+
+export class Coordinate {
+  public row: Coordinates['row'];
+
+  public column: Coordinates['column'];
+
+  public coordinateSystem: CoordinateSystem;
+
+  constructor(
+    row: Coordinates['row'],
+    column: Coordinates['column'],
+    coordinateSystem: CoordinateSystem,
+  ) {
+    if (!coordinateSystem.rows().includes(row)) {
+      throw new Error(
+        `Expected a row with value of ${coordinateSystem
+          .rows()
+          .join(',')}, got ${row}.`,
+      );
+    }
+    this.row = row;
+
+    if (!coordinateSystem.columns().includes(column)) {
+      throw new Error(
+        `Expected a column with value of ${coordinateSystem
+          .columns()
+          .join(',')}, got ${column}.`,
+      );
+    }
+    this.column = column;
+
+    this.coordinateSystem = coordinateSystem;
+  }
+
+  static fromString(
+    coordinateString: string,
+    coordinateSystem: CoordinateSystem,
+  ): Coordinate {
+    const rows = coordinateSystem.rows();
+    const rowsOptions = rows.join('|');
+
+    const columns = [
+      ...coordinateSystem.columns(),
+      ...coordinateSystem.paddedColumns(),
+    ];
+    const columnsOptions = columns.join('|');
+
+    const matches = coordinateString.match(
+      new RegExp(`^(${rowsOptions})(${columnsOptions})`),
+    );
+
+    if (matches === null || matches?.length === 0) {
+      const firstValidExample = rows[0].toString() + columns[0].toString();
+      const lastValidExample =
+        rows[rows.length - 1].toString() +
+        columns[columns.length - 1].toString();
```

Comment:
Similar here: there should be a null-check for this case. You can use Lodash's `last` function to have a type-safe alternative here.